File: common_native_lib/bilog.cpp
/*
 * Yuchuan Sun
 * Version 1.0
 * Date 2016-8-5
 * BiLog common module that is used by native code
*/

#include "logPrint.h"
#define LOG_TAG "BILog"
#include <utils/Log.h>
#include <sys/time.h>
#include <string.h>
#include <stdio.h>

#include "bilog.h"

static const char *log_version = "02";

BiLog::BiLog(const char *logType):
    mLogHeader(new Json::Value),
    mParam(new Json::Value),
    mParams(new Json::Value)
{
    ALOGI("constructor");
    struct timeval time;
    long long happenTime = 0;
    char strHappenTime[128] = {0};
    gettimeofday(&time,NULL);
    happenTime = ((long long)time.tv_sec*1000 + (long long)time.tv_usec/1000);
    snprintf(strHappenTime,sizeof(strHappenTime),"%lld",happenTime);

    PutHeader("logVersion",log_version);
    PutHeader("logType",logType);
    //PutHeader("happenTime",strHappenTime);
    PutHeader("happenTime",happenTime);
}

BiLog::~BiLog()
{
    ALOGI("destructor");
    if(mLogHeader)
        delete mLogHeader;
    mLogHeader = NULL;
    if(mParam)
        delete mParam;
    mParam = NULL;
    if(mParams)
        delete mParams;
    mParams = NULL;
}

void BiLog::PutHeader(const char * name, const char * value)
{
    (*(mLogHeader))[name] = value;
}

void BiLog::PutHeader(const char * name, long long value)
{
    (*(mLogHeader))[name] = value;
}

void BiLog::PushParameter(const char * name, const char * value)
{
    (*(mParam))[name] = value;
    //mParams.append(mParam);
}

const char * BiLog::GetString()
{
    (*(mLogHeader))["params"] = *(mParam);
    ALOGD("%s",mLogHeader->toStyledString().c_str());
    return mLogHeader->toStyledString().c_str();
}

File: common_java_lib/common/src/com/whaley/biutil/BiLog.java
package com.whaley.biutil;

import org.json.JSONObject;

/**
 * Created by wanglei on 2016/7/28.
 */
public class BiLog implements IBiLog {
    private String LogVersion = "02";
    private JSONObject LogHeader = new JSONObject();
    protected JSONObject Params = new JSONObject();

    protected void PutHeader(String name, String value) {
	try {
	    LogHeader.put(name, value);
	}
	catch (Exception e) {
	}
    }

    protected void PutHeader(String name, long value) {
	try {
	    LogHeader.put(name, value);
	}
	catch (Exception e) {
	}
    }
    
    public BiLog(String logType) {
	PutHeader("logVersion", LogVersion);
	PutHeader("logType", logType);
	PutHeader("happenTime", System.currentTimeMillis());
    }
    
    @Override
    public void PushParameter(String name, String value)
    {
	try {
	    Params.put(name, value);
	}
	catch (Exception e) {
	}
    }

    @Override
    public String GetString()
    {
	String logstr = null;
	try {
	    JSONObject jsonLog = new JSONObject(LogHeader.toString());
	    jsonLog.put("params", Params);
	    logstr = jsonLog.toString();
	}
	catch (Exception e) {
	}
        return logstr;
    }
}
